File: Framework/CSharp/Framework/Framework/SmartAssert.cs
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System;
using System.Diagnostics;

namespace Smartkernel.Framework
{
    /// <summary>
    /// 断言
    /// </summary>
    public class SmartAssert
    {
        /// <summary>
        /// 判断是否为True，不为True则抛出异常
        /// </summary>
        /// <param name="input">待判断项目</param>
        public static void IsTrue(bool input)
        {
            if (!input)
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }

        /// <summary>
        /// 判断是否为False，不为False则抛出异常
        /// </summary>
        /// <param name="input">待判断项目</param>
        public static void IsFalse(bool input)
        {
            if (input)
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }

        /// <summary>
        /// 判断是否相等，不相等则抛出异常
        /// </summary>
        /// <param name="left">左对象</param>
        /// <param name="right">右对象</param>
        public static void IsEquals(object left, object right)
        {
            if (!left.Equals(right))
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }

        /// <summary>
        /// 判断是否不相等，相等则抛出异常
        /// </summary>
        /// <param name="left">左对象</param>
        /// <param name="right">右对象</param>
        public static void IsNotEquals(object left, object right)
        {
            if (left.Equals(right))
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }

        /// <summary>
        /// 判断是否为Null，不为Null则抛出异常
        /// </summary>
        /// <param name="input">待判断项目</param>
        public static void IsNull(object input)
        {
            if (input != null)
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }

        /// <summary>
        /// 判断是否不为Null，为Null则抛出异常
        /// </summary>
        /// <param name="input">待判断项目</param>
        public static void IsNotNull(object input)
        {
            if (input == null)
            {
                var stackFrame = new StackFrame(true);
                throw new Exception(stackFrame.ToString());
            }
        }
    }
}

File: Framework/Php/framework/framework/smartassert.php
<?php

/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */

namespace Smartkernel\Framework {

    use \Exception;

/*
     * 断言
     */

    class SmartAssert {
        /*
         * 判断是否为True，不为True则抛出异常
         * 
         * @param bool $input，待判断项目
         * 
         */

        public static function isTrue($input) {
            if ($input === false) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

        /*
         * 判断是否为False，不为False则抛出异常
         * 
         * @param bool $input，待判断项目
         * 
         */

        public static function isFalse($input) {
            if ($input === true) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

        /*
         * 判断是否相等，不相等则抛出异常
         * 
         * @param object $left，左对象
         * @param object $right，右对象
         * 
         */

        public static function isEquals($left, $right) {
            if ($left !== $right) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

        /*
         * 判断是否不相等，相等则抛出异常
         * 
         * @param object $left，左对象
         * @param object $right，右对象
         * 
         */

        public static function isNotEquals($left, $right) {
            if ($left !== $right) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

        /*
         * 判断是否为Null，不为Null则抛出异常
         * 
         * @param bool $input，待判断项目
         * 
         */

        public static function isNull($input) {
            if ($input !== null) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

        /*
         * 判断是否不为Null，为Null则抛出异常
         * 
         * @param bool $input，待判断项目
         * 
         */

        public static function isNotNull($input) {
            if ($input === null) {
                throw new Exception(var_export(debug_backtrace()));
            }
        }

    }

}
